### ISSUE TITLE
Ensure every ARIA input field has an accessible name

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-core#2505]
# Bug Report

## Description
A new version of axe-core (3.3.0) has detected an accessibility error around how the select is marked up. 

## Steps to Reproduce

1. Run WDIO tests with axe-core 3.3.0
2. View error output for terra-select accessibility tests

## Additional Context / Screenshots
Error output:
```
expected no accessibility violations but got:
        {
  "description": "Ensures every ARIA input field has an accessible name",
  "help": "ARIA input fields have an accessible name",
  "helpUrl": "https://dequeuniversity.com/rules/axe/3.3/aria-input-field-name?application=axeAPI",
  "id": "aria-input-field-name",
  "impact": "serious",
  "nodes": [
    {
      "all": [],
      "any": [
        {
          "data": null,
          "id": "aria-label",
          "impact": "serious",
          "message": "aria-label attribute does not exist or is empty",
          "relatedNodes": []
        },
        {
          "data": null,
          "id": "aria-labelledby",
          "impact": "serious",
          "message": "aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty",
          "relatedNodes": []
        },
        {
          "data": null,
          "id": "non-empty-title",
          "impact": "serious",
          "message": "Element has no title attribute or the title attribute is empty",
          "relatedNodes": []
        }
      ],
      "failureSummary": "Fix any of the following:\n  aria-label attribute does not exist or is empty\n  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty\n  Element has no title attribute or the title attribute is empty",
      "html": "<div class=\"_Frame-module__display___8mHS5\" role=\"textbox\"><div id=\"terra-select-screen-reader-placeholder-23\" class=\"_Frame-module__placeholder___Lfgel\">Select a size</div></div>",
      "impact": "serious",
      "none": [],
      "target": [
        "._Frame-module__is-focused___MU4rY > ._Frame-module__display___8mHS5[role=\"textbox\"]"
      ]
    },
    {
      "all": [],
      "any": [
        {
          "data": null,
          "id": "aria-label",
          "impact": "serious",
          "message": "aria-label attribute does not exist or is empty",
          "relatedNodes": []
        },
        {
          "data": null,
          "id": "aria-labelledby",
          "impact": "serious",
          "message": "aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty",
          "relatedNodes": []
        },
        {
          "data": null,
          "id": "non-empty-title",
          "impact": "serious",
          "message": "Element has no title attribute or the title attribute is empty",
          "relatedNodes": []
        }
      ],
      "failureSummary": "Fix any of the following:\n  aria-label attribute does not exist or is empty\n  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty\n  Element has no title attribute or the title attribute is empty",
      "html": "<ul id=\"terra-select-menu\" role=\"listbox\" class=\"_Menu-module__menu___2E516\" aria-activedescendant=\"terra-select-option-Red\" tabindex=\"0\">",
      "impact": "serious",
      "none": [],
      "target": [
        "#terra-select-menu"
      ]
    }
  ],
  "tags": [
    "wcag2a",
    "wcag412"
  ]
}
```

## Expected Behavior
Components are free of accessibility violations to the best of our knowledge.

## Possible Solution
Add aria-label to DOM element that has role="textbox"

## Environment

* Component Name and Version:
* Browser Name and Version:
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile):
